akka-projection-core/src/test/scala/akka/projection/internal/metrics/tools/InMemTelemetry.scala

```diff
@@ -32,8 +32,11 @@ class InMemTelemetry(projectionId: ProjectionId, system: ActorSystem[_]) extends
   override def stopped(): Unit =
     stoppedInvocations.incrementAndGet()
 
-  override def afterProcess(readyTimestampNanos: Long): Unit = {
+  override def beforeProcess(): AnyRef = ExternalContext()
+
+  override def afterProcess(externalContext: AnyRef): Unit = {
     afterProcessInvocations.incrementAndGet()
+    val readyTimestampNanos = externalContext.asInstanceOf[ExternalContext].readyTimestampNanos
     lastServiceTimeInNanos.set(System.nanoTime() - readyTimestampNanos)
   }
```

Comment:
the `InMemTelemetry` provides an example of an extranlly-provided context. It measures the time between `beforeProcess` and `afterProcess`.